dependencies/com.b2international.org.apache.lucene/src/com/b2international/index/es/EsClient.java

```diff
@@ -164,8 +176,39 @@ private EsClient(final EsClientConfiguration configuration) {
 				.setRequestConfigCallback(requestConfigCallback)
 				.setMaxRetryTimeoutMillis(configuration.getSocketTimeout()); // retry timeout should match socket timeout
 			
+			boolean useAuthentication = !Strings.isNullOrEmpty(configuration.getUserName()) && !Strings.isNullOrEmpty(configuration.getPassword());
+
+			if (useAuthentication) {
+				
+				final HttpClientConfigCallback httpClientConfigCallback = httpClientConfigBuilder -> {
+					final BasicCredentialsProvider credentialsProvider = new BasicCredentialsProvider();
+					credentialsProvider.setCredentials(AuthScope.ANY, 
+							new UsernamePasswordCredentials(configuration.getUserName(), configuration.getPassword()));
+					return httpClientConfigBuilder.setDefaultCredentialsProvider(credentialsProvider);
+				};
+				
+				restClientBuilder.setHttpClientConfigCallback(httpClientConfigCallback);
+				
+			}
+			
+			LOG.info("ES REST client is connecting to '{}'{}, connect timeout: {} ms, socket timeout: {} ms.", 
+					host.toURI(),
+					useAuthentication ? " using basic authentication" : "",
+					configuration.getConnectTimeout(),
+					configuration.getSocketTimeout());
+			
 			final RestHighLevelClient client = new RestHighLevelClient(restClientBuilder);
-			checkState(client.ping(), "The cluster at '%s' is not available.", host.toURI());
+			
+			try {
+				checkState(client.ping(), "The cluster at '%s' is not available.", host.toURI());
+			} catch (Exception e) {
+				if (e instanceof ElasticsearchStatusException && ((ElasticsearchStatusException) e).status() == RestStatus.UNAUTHORIZED) {
+					LOG.error("Unable to authenticate with remote cluster '{}' using the predefined credentials", host.toURI());
```

Comment:
-predefined +given? "Predefined" could make administrators anxious, they might think we are in control of their passwords.